NetTally/ReorderTasksWindow.xaml.cs

```diff
@@ -0,0 +1,125 @@
+
+            MainViewModel.PropertyChanged += MainViewModel_PropertyChanged;
+
+            // Create filtered, sortable views into the collection for display in the window.
+            if (MainViewModel.TaskList != null)
+            {
+                TaskView = new ListCollectionView(MainViewModel.TaskList);
+            }
+
+            // Initialize starting selected positions
+            TaskView.MoveCurrentToPosition(-1);
+
+            // Set the data context for binding.
+            DataContext = this;
+
+            // Update the displayed order immediately
+            UpdateTaskList();
+        }
+        #endregion
+
+        private void UpdateTaskList()
+        {
+            VoteCounter.Instance.OrderedTaskList = TaskView.SourceCollection as List<string>;
+            OnPropertyChanged("Votes");
+            TaskView.Refresh();
+        }
+
+        #region INotifyPropertyChanged implementation
+        /// <summary>
+        /// Event for INotifyPropertyChanged.
+        /// </summary>
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <summary>
+        /// Function to raise events when a property has been changed.
+        /// </summary>
+        /// <param name="propertyName">The name of the property that was modified.</param>
+        protected void OnPropertyChanged([CallerMemberName] string propertyName = null)
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+        }
+        #endregion
+
+        #region Watched Events        
+        /// <summary>
+        /// Watch for notifications from the main view model about changes in the vote backend.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="PropertyChangedEventArgs"/> instance containing the event data.</param>
+        private void MainViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == "Votes")
+            {
+                TaskView.Refresh();
+            }
+        }
+
+        private void up_Click(object sender, RoutedEventArgs e)
+        {
+            MainViewModel.TaskList.Swap(TaskView.CurrentPosition, TaskView.CurrentPosition - 1);
+            UpdateTaskList();
+        }
+
+        private void down_Click(object sender, RoutedEventArgs e)
+        {
+            MainViewModel.TaskList.Swap(TaskView.CurrentPosition, TaskView.CurrentPosition + 1);
+            UpdateTaskList();
+        }
+
+        private void alpha_Click(object sender, RoutedEventArgs e)
+        {
+            MainViewModel.TaskList.Sort();
+            UpdateTaskList();
+        }
+
+        private void default_Click(object sender, RoutedEventArgs e)
+        {
+            MainViewModel.TaskList.Sort((a,b) => (MainViewModel.KnownTasks.ToList().IndexOf(a).CompareTo(MainViewModel.KnownTasks.ToList().IndexOf(b))));
```

Comment:
This sorting algorithm is really bad.  It will re-enumerate `KnownTasks` (which re-enumerate the base vote lists) twice for every single element in the list.

And, in fact, the end result is a list in the exact order as the `KnownTasks` enumerable, which makes the sorting redundant anyway.

Change to:
```
MainViewModel.TaskList.Clear();
MainViewModel.TaskList.AddRange(MainViewModel.KnownTasks);
```
The local TaskView will update along with it, so that `UpdateTaskList` works fine.